File: solutions/c/perfect-numbers/1/perfect_numbers.c
#include "perfect_numbers.h"


kind classify_number(int number) {
    int isFactor = 1;
    int aliquot = 0;
    
    //Checks at the beginning if the number is valid positive integer else returns error
    if ( !number || number < 0 ) {
      return ERROR;
    }
        
    while (isFactor != number) {

        if (number % isFactor == 0) {

            aliquot += isFactor;    
        }
        isFactor++;
    }


    //Checks for PERFECT, ABUNDANT and DEFICIENT kind ---> returns kind.
    if (aliquot == number) {
        return PERFECT_NUMBER;
        }
    if (aliquot < number) {
        return DEFICIENT_NUMBER;
        }
    if (aliquot > number) {
        return ABUNDANT_NUMBER;
        }

   //Catches an error if all the conditions above are not met for some reason.
   return ERROR;
}

File: solutions/python/perfect-numbers/1/perfect_numbers.py
def classify(number):
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """
    if (number <= 0):
        raise ValueError("Classification is only possible for positive integers.")
    aliquot_sum = sum(value for value in range(1,number) if number % value == 0) 
    if aliquot_sum == number:
        return "perfect"
    if aliquot_sum >= number:
        return "abundant"
    return "deficient"